src/main/java/roboy/dialog/states/ordinaryStates/FUAState.java

```diff
@@ -0,0 +1,103 @@
+package roboy.dialog.states.ordinaryStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.context.contextObjects.IntentValue;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jRelationships;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.MemoryNodeModel;
+import roboy.util.QAJsonParser;
+import roboy.util.RandomList;
+
+import java.util.List;
+import java.util.Set;
+
+import static roboy.memory.Neo4jRelationships.*;
+
+/**
+ * Follow Up Asking State
+ * This state is only entered if there are some known facts about the active interlocutor.
+ * It asks if the known facts are still up to date.
+ */
+public class FUAState extends State {
+
+    private QAJsonParser qaValues;
+    private Neo4jRelationships[] predicates = { FROM, HAS_HOBBY, WORK_FOR, STUDY_AT };
+    private Neo4jRelationships selectedPredicate;
+    private State nextState;
+
+    private final String TRANSITION_INFO_UPDATED = "questionAnswering";
+    private final Logger LOGGER = LogManager.getLogger();
+
+    public FUAState(String stateIdentifier, StateParameters params) {
+        super(stateIdentifier, params);
+        qaValues = new QAJsonParser(params.getParameter("qaFile"));
+    }
+
+    @Override
+    public State.Output act() {
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+
+        for (Neo4jRelationships predicate : predicates) {
+            if (person.hasRelationship(predicate) && !Context.getInstance().DIALOG_INTENTS.contains(new IntentValue("FUP", predicate))) {
```

Comment:
Long line, a line break wouldn't hurt. Please don't hate me for having a small monitor :)

Reply:
Done